velox/expression/tests/CastExprTest.cpp

```diff
@@ -921,3 +921,51 @@ TEST_F(CastExprTest, castAsCall) {
   auto expected = makeNullableFlatVector(outputValues);
   assertEqualVectors(expected, result);
 }
+
+namespace {
+/// Wraps input in a constant encoding that repeats the first element and then
+/// in dictionary that reverses the order of rows.
+class TestingDictionaryOverConstFunction : public exec::VectorFunction {
```

Comment:
Thank you for adding this test! 

nit: Consider moving this vector function to a separate file under velox/vector/tests so that it can be reused by other tests later. There are two similar vector function there.

Reply:
This sounds good to me. Thanks!